src/main/java/org/traccar/database/DeviceManager.java

```diff
@@ -117,13 +117,7 @@ public Device getByUniqueId(String uniqueId) throws SQLException {
 
     @Override
     public String getDevicePassword(long id, String defaultPassword) {
-
-        String password = getById(id).getString(Command.KEY_DEVICE_PASSWORD);
-        if (password != null) {
-            return password;
-        }
-
-        return defaultPassword;
+        return lookupAttributeString(id, Command.KEY_DEVICE_PASSWORD, defaultPassword, false);
```

Comment:
I think it should be configurable in the configuration, not server. That's for security reasons.

Also, I think it should be configurable per protocol.

Reply:
Sorry, commented bellow...

Reply:
How does your version solve any of the mentioned problems?

Reply:
You can create a group of devices, lets say all same model devices, or all devices of a specific user, even if not the same protocol, then add the `devicePassword` attribute for that group, and then delegate management of that group to an specific administrator.

Reply:
Maybe the attribute can be named `protocol.devicePassword` to make it a protocol specific, server wide attribute, but I think the proposed approach is more generic.

Reply:
As you said, it can be a file configuration, in the same form `protocol.devicePassword`, but again, the proposed approach I think is more generic. Thanks.

Reply:
I think there is some misunderstanding. Your approach is already server-wide.

Reply:
> Maybe the attribute can be named `protocol.devicePassword` to make it a protocol specific, server wide attribute, but I think the proposed approach is more generic.

Yes, I meant we can implement it only as a per protocol, server wide attribute, instead of a server wide attribute for all protocols. But, sorry for repetition, the proposed approach is not restricted to a protocol, you can make it as wide as you like using device groups, and solves the problem that for now, we need to define a per device password if we don't want to use the hard-coded protocol default. Am I wrong about the last part? Is that what you are trying to say?

Reply:
Of course, for this to work we need to enable `deviceManager.lookupGroupsAttribute` in config

Reply:
I'm not saying that it shouldn't be configurable per group. I'm saying that it also should be configurable per protocol and in the config file. It's more flexible than your approach.

Reply:
Oh, I see. So, are you saying that we must keep this approach I'm proposing, but also we need to make it configurable via `traccar.xml` using somethink like `[protocol].devicePassword`?

Reply:
Yes.

Reply:
Great! We are now on the same page. I'll come back. Thank you very much.